internal/analytics/analytics.go

```diff
@@ -0,0 +1,155 @@
+package analytics
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/auth0/auth0-cli/internal/buildinfo"
+	"github.com/spf13/cobra"
+)
+
+const (
+	eventNamePrefix   = "CLI"
+	analyticsEndpoint = "https://heapanalytics.com/api/track"
+	appID             = "1279799279"
+	versionKey        = "version"
+	osKey             = "os"
+	archKey           = "arch"
+)
+
+type Tracker struct {
+	wg sync.WaitGroup
+}
+
+type event struct {
+	App        string            `json:"app_id"`
+	ID         string            `json:"identity"`
+	Event      string            `json:"event"`
+	Timestamp  int64             `json:"timestamp"`
+	Properties map[string]string `json:"properties"`
+}
+
+func NewTracker() *Tracker {
+	return &Tracker{}
+}
+
+func (t *Tracker) TrackFirstLogin(id string) {
+	eventName := fmt.Sprintf("%s - Auth0 - First Login", eventNamePrefix)
+	t.track(eventName, id)
+}
+
+func (t *Tracker) TrackCommandRun(cmd *cobra.Command, id string) {
+	eventName := generateRunEventName(cmd.CommandPath())
+	t.track(eventName, id)
+}
+
+func (t *Tracker) Wait(ctx context.Context) {
+	ch := make(chan struct{})
+
+	go func() {
+		t.wg.Wait()
+		close(ch)
+	}()
+
+	select {
+	case <-ch: // waitgroup is done
+		return
+	case <-ctx.Done():
+		return
+	}
+}
+
+func (t *Tracker) track(eventName string, id string) {
+	if !shouldTrack() {
+		return
+	}
+
+	event := newEvent(eventName, id)
+
+	t.wg.Add(1)
```

Comment:
Just a commentary that we can't technically do `wg.Add` _after_ we do `wg.Wait` -- that would cause a panic. As long as we're synchronously done doing all the `wg.Add`s before we Wait, it's fine.

There's other async style ways we can do this as well (e.g. utilizing a single channel,  a single worker) to future proof against this potential issue.